apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/exception/Exceptions.kt

```diff
@@ -98,11 +98,19 @@ class CacheMissException(val key: String, val fieldName: String? = null) : Apoll
 class HttpCacheMissException(message: String, cause: Exception? = null) : ApolloException(message = message, cause = cause)
 
 /**
- * Multiple exceptions happened, for an exemple with a [CacheFirst] fetch policy
+ * Multiple exceptions happened, for an example with a [CacheFirst] fetch policy
+ * [cause] might change to null and second exception will also be added as suppressed exception
  */
 class ApolloCompositeException(first: Throwable?, second: Throwable?) : ApolloException(message = "multiple exceptions happened", second) {
+
   val first = (first as? ApolloException) ?: throw RuntimeException("unexpected first exception", first)
+
   val second = (second as? ApolloException) ?: throw RuntimeException("unexpected second exception", second)
+
+  init {
+    if (first != null && second != null) addSuppressed(first)
```

Comment:
Ok to keep `second` for the cause, but to add both `first` and `second` to the suppressed? So something like this:

```suggestion
    if (first != null) addSuppressed(first)
    if (second != null) addSuppressed(second)
```

Reply:
then there will be a duplicate stack trace like
1. SecondException
2. FirstException
3. SecondException

Reply:
Oh ok didn't realize they would appear in the stack trace by default!

I'm not sure it's really an issue though? Maybe it's ok to have the `second` one appearing twice (once for cause, and once for suppressed), at least you can't miss it 😅

Reply:
Okay, that way should i also add previously suggested deprecation on first and second usages?

Reply:
Yes that would be nice!

Reply:
+1 to both suppressed for consistency and symmmetry. Also to make sure we have `second` when we will remove it from the `cause` later on.